src/toolbox.ts

```diff
@@ -2125,201 +2331,243 @@ export class FilterPointDistanceFromRow extends ToolboxItem {
+        $("#filter-single-class-mode").toggleClass("ulabel-hidden");
+        $("#filter-multi-class-mode").toggleClass("ulabel-hidden");
     }
 
     /**
      * Toggles whether or not the options should be displayed.
      */
     private toggleCollapsedOptions() {
         // Toggle the class which collapses the options
-        $("fieldset.filter-row-distance-options").toggleClass("ulabel-collapsed")
+        $("fieldset.filter-row-distance-options").toggleClass(
+            "ulabel-collapsed",
+        );
 
         // Toggle the state
-        this.collapse_options = !this.collapse_options
+        this.collapse_options = !this.collapse_options;
 
         // Save the state to the user's browser so it can be re-loaded in the same state
-        set_local_storage_item("filterDistanceCollapseOptions", this.collapse_options)
+        set_local_storage_item(
+            "filterDistanceCollapseOptions",
+            this.collapse_options,
+        );
     }
 
     private create_overlay() {
         // Get only the set of all line annotations
-        const line_annotations: ULabelAnnotation[] = get_point_and_line_annotations(this.ulabel)[1]
+        const line_annotations: ULabelAnnotation[] =
+            get_point_and_line_annotations(this.ulabel)[1];
 
         // Initialize an object to hold the distances points are allowed to be from each class as well as any line
-        let filter_values: DistanceFromPolylineClasses = {closest_row: undefined}
+        const filter_values: DistanceFromPolylineClasses = {
+            closest_row: undefined,
+        };
 
         // Grab all filter-distance-sliders on the page
-        const sliders: NodeListOf<HTMLInputElement> = document.querySelectorAll(".filter-row-distance-slider")
+        const sliders: NodeListOf<HTMLInputElement> = document.querySelectorAll(
+            ".filter-row-distance-slider",
+        );
 
         // Loop through each slider and populate filter_values
         for (let idx = 0; idx < sliders.length; idx++) {
             // Use a regex to get the string after the final - character in the slider id (Which is the class id or the string "closest_row")
-            const slider_class_name = /[^-]*$/.exec(sliders[idx].id)[0]
+            const slider_class_name = /[^-]*$/.exec(sliders[idx].id)[0];
 
             // Use the class id as a key to store the slider's value
             filter_values[slider_class_name] = {
-                distance: sliders[idx].valueAsNumber
-            }
+                distance: sliders[idx].valueAsNumber,
+            };
         }
 
         // Create and assign an overlay class instance to ulabel to be able to access it
         this.overlay = new FilterDistanceOverlay(
             this.ulabel.config["image_width"] * this.ulabel.config["px_per_px"],
-            this.ulabel.config["image_height"] * this.ulabel.config["px_per_px"],
+            this.ulabel.config["image_height"] *
+                this.ulabel.config["px_per_px"],
```

Comment:
This is a particularly ugly case of line wrapping. The [printWidth](https://prettier.io/docs/en/options#print-width) option in `prettier` claims it's just a target line length (not a hard limit), and [`max-len`](https://eslint.org/docs/latest/rules/max-len) in ESLint sets a hard limit (although it's deprecated, so if we want that we should use [`max-len` from `@stylistic/eslint-plugin-js`](https://eslint.style/rules/js/max-len)).

I've only ever deviated from line length 80 for personal or internal projects, so I'd lean on the side of keeping 80 here (and maybe adding that ESLint plugin to enforce it). We'll have some spots like this where we need to clean up the readability as a result, but I think that's fine. Thoughts?

Reply:
yeah id prefer the 80 line length rather than whatever logic prettier uses. looking at some of the other spots (like that for loop expansion) im starting to lean towards just using eslint and not prettier

Reply:
ESLint doesn't do any formatting - it can auto-fix a few issues it reports but it won't enforce code style very much. 

VSCode has a Typescript formatter baked-in (see [this](https://code.visualstudio.com/docs/languages/typescript#_formatting)) which is more reasonable, but I'm not sure what it's actually calling under the hood, or how to get it as a git hook.

Reply:
From the end of [this](https://github.com/microsoft/TypeScript/blob/ca18009b8bd5cea6a7fbcba3d7e4dd3c4633bc92/src/services/formatting/README.md) document:
> The formatter is not exported publicly, and so all usage can only come through the language server.
As much as I'd love to fork TypeScript just to get the formatter, it's probably smarter to pick an alternative.

Reply:
I've looked at a few, most places seem to recommend `prettier`.
[typescript-formatter](https://github.com/vvakame/typescript-formatter) is extremely minimal, not enough for us I think
[gts](https://github.com/google/gts) from google just uses `prettier` under the hood :roll_eyes: 
[@stylistic/eslint-plugin](https://eslint.style/packages/ts) really looks like it's what we want - it has a unified plugin for JS and TS, with toggles for all rules, that doesn't immediately obliterate all of our code.

Most of the stuff `prettier` is obliterating is problematic in the first place, but if we can avoid a massive style refactor and instead do some incremental improvements, I would much prefer that.

Reply:
let's rerun here and just resolve the outstanding comments